llvm/include/llvm/CodeGen/MachineInstr.h

```diff
@@ -1725,6 +1730,10 @@ class MachineInstr
   /// the instruction's location and its intended destination.
   bool isSafeToMove(bool &SawStore) const;
 
+  /// Return true if this instruction can be considered dead if all of its
+  /// defined registers are dead.
+  bool canBeDead() const;
```

Comment:
In IR the corresponding query is called wouldInstructionBeTriviallyDead(). I'd maybe use `wouldBeTriviallyDead()` here for symmetry.

Reply:
Done. Thanks!